core/rawdb/database.go

```diff
@@ -341,46 +382,67 @@ func InspectDatabase(db ethdb.Database) error {
 		}
 	}
 	// Inspect append-only file store then.
-	ancients := []*common.StorageSize{&ancientHeaders, &ancientBodies, &ancientReceipts, &ancientHashes, &ancientTds}
+	ancientSizes := []*common.StorageSize{&ancientHeadersSize, &ancientBodiesSize, &ancientReceiptsSize, &ancientHashesSize, &ancientTdsSize}
 	for i, category := range []string{freezerHeaderTable, freezerBodiesTable, freezerReceiptTable, freezerHashTable, freezerDifficultyTable} {
 		if size, err := db.AncientSize(category); err == nil {
-			*ancients[i] += common.StorageSize(size)
+			*ancientSizes[i] += common.StorageSize(size)
 			total += common.StorageSize(size)
 		}
 	}
+	// Get number of ancient rows inside the freezer
+	ancients := counter(0)
+	if count, err := db.Ancients(); err == nil {
+		ancients = counter(count)
+	}
+	// Count receipts in ancient db
+	ancientReceipts := counter(0)
+	for blockNumber := uint64(0); blockNumber < uint64(ancients); blockNumber++ {
+		data, err := db.Ancient(freezerReceiptTable, blockNumber)
+		if err != nil {
+			log.Error("Error reading ancient receipts from block", "number", blockNumber, "err", err)
+		} else {
+			ancientReceipts += countReceiptsRLP(data)
+		}
+		if blockNumber%1000 == 0 && time.Since(logged) > 8*time.Second {
+			log.Info("Inspecting ancient database", "number", blockNumber, "elapsed", common.PrettyDuration(time.Since(start)))
```

Comment:
Would be nice to print like `Inspecting ancient database for receipts counting`.
And also we can print the total blocks needs to be iterated so that it's kind of progress marker

Reply:
I created a percentage to have a better feeling of the progress. For the message, `Inspecting ancient database for receipts counting` goes over the formatting of other messages:

```zsh
INFO [08-28|05:23:49.768] Loaded most recent local header          number=3301083 hash="899a9a…3d2961" td=4817450 age=23m4s
INFO [08-28|05:23:49.768] Loaded most recent local full block      number=3301083 hash="899a9a…3d2961" td=4817450 age=23m4s
INFO [08-28|05:23:49.768] Loaded most recent local fast block      number=3301083 hash="899a9a…3d2961" td=4817450 age=23m4s
INFO [08-28|05:23:49.768] Loaded last fast-sync pivot marker       number=3300981
INFO [08-28|05:23:57.834] Inspecting database                      count=19601000 elapsed=8.000s
INFO [08-28|05:24:05.834] Inspecting ancient database for receipts counting number=373000 percentage=11 elapsed=16.000s
INFO [08-28|05:24:13.842] Inspecting ancient database for receipts counting number=1571000 percentage=47 elapsed=24.007s
INFO [08-28|05:24:21.845] Inspecting ancient database for receipts counting number=2042000 percentage=62 elapsed=32.010s
INFO [08-28|05:24:29.860] Inspecting ancient database for receipts counting number=2448000 percentage=74 elapsed=40.025s
```

What do you think of `Counting ancient database receipts`? Does it make sense in english?

```zsh
INFO [08-28|05:32:53.088] Loaded most recent local header          number=3301083 hash="899a9a…3d2961" td=4817450 age=32m8s
INFO [08-28|05:32:53.088] Loaded most recent local full block      number=3301083 hash="899a9a…3d2961" td=4817450 age=32m8s
INFO [08-28|05:32:53.088] Loaded most recent local fast block      number=3301083 hash="899a9a…3d2961" td=4817450 age=32m8s
INFO [08-28|05:32:53.088] Loaded last fast-sync pivot marker       number=3300981
INFO [08-28|05:33:01.199] Inspecting database                      count=19700000 elapsed=8.044s
INFO [08-28|05:33:09.206] Counting ancient database receipts       number=628000  percentage=19 elapsed=16.051s
INFO [08-28|05:33:17.223] Counting ancient database receipts       number=1694000 percentage=51 elapsed=24.068s
INFO [08-28|05:33:25.227] Counting ancient database receipts       number=2108000 percentage=64 elapsed=32.072s
INFO [08-28|05:33:33.232] Counting ancient database receipts       number=2529000 percentage=77 elapsed=40.078s
INFO [08-28|05:33:41.243] Counting ancient database receipts       number=2865000 percentage=87 elapsed=48.088s
INFO [08-28|05:33:49.257] Counting ancient database receipts       number=3111000 percentage=94 elapsed=56.103s
```

Reply:
Yes, it looks good to me. Maybe it would be better to add the 'final' state: `Counting ancient database receipts       number=3111000 percentage=100 elapsed=56.103s`

Reply:
Done.